### PR TITLE
Python debugging support using ptvsd

### PR DESCRIPTION
**Pull request by anton-st**

### CODE REVIEW

File: pkg/debuggers/python/impl.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		fmt.Sscanf(pe, "%d", &port)
+		return port, nil
+	}
+
+	root := filepath.Join("/proc", fmt.Sprintf("%d", pid), "cwd")
+	log.WithField("root", root).Debug("searching root")
+	re := regexp.MustCompile(PtvsdSearchString)
+
+	filepath.Walk(root+"/", func(p string, fi os.FileInfo, err error) error {
+		if err != nil || fi == nil || fi.IsDir() {
```
Comment (yuval-k) [inline pkg/debuggers/python/impl.go]: consider checking just for *py files?

File: pkg/debuggers/python/impl.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+			return nil
+		}
+
+		b, ferr := ioutil.ReadFile(p)
+		if ferr != nil {
+			return nil
+		}
+
+		s := re.FindString(string(b))
```
Comment (yuval-k) [inline pkg/debuggers/python/impl.go]: reading a large file might consume a lot of memory - consider  using FindReaderIndex

File: pkg/debuggers/python/impl.go (hunk 1 above)
Comment (anton-st) [reply pkg/debuggers/python/impl.go]: I thought about it and I'll probably do it, but than: http://dcjtech.info/topic/python-file-extensions/
I can also try to ignore known compiled files, like pyc, pyo etc. instead.

File: pkg/debuggers/python/impl.go (hunk 2 above)
Comment (anton-st) [reply pkg/debuggers/python/impl.go]: I'll need to understand that - let's talk offline...

**Comment by anton-st** [comment]
Implemented all changes requested. Please review.

File: pkg/debuggers/python/impl.go [hunk 3]
```diff
@@ -72,17 +75,30 @@ func getPtvsdPort(pid int) (int, error) {
...
+
+		idxs := re.FindReaderIndex(bufio.NewReader(io.LimitReader(f, PtvsdMaxFileSize)))
+		if idxs == nil {
+			fileNum++
+			if fileNum >= PtvsdMaxNumberOfFiles {
+				return errors.New("File limit reached")
+			}
+			return nil
+		}
+		f.Seek(int64(idxs[0]), 0)
```
Comment (yuval-k) [inline pkg/debuggers/python/impl.go]: use constant SeekStart

File: pkg/debuggers/python/impl.go [hunk 4]
```diff
@@ -72,17 +75,30 @@ func getPtvsdPort(pid int) (int, error) {
...
+		b := make([]byte, idxs[1]-idxs[0])
+		_, err = f.Read(b)
+		if err != nil {
+			return nil
+		}
+		args := strings.Split(string(b), ",")
 		if len(args) > 2 {
 			fmt.Sscanf(args[2], "%d", &port)
```
Comment (yuval-k) [inline pkg/debuggers/python/impl.go]: err is ignored here

File: pkg/debuggers/python/impl.go [hunk 5]
```diff
@@ -55,7 +58,7 @@ func (i *PythonInterface) Attach(pid int) (debuggers.DebugServer, error) {
 // Search /proc/{PID}/cwd for file with "ptvsd.enable_attach" string and extracts port from it
```
Comment (yuval-k) [inline pkg/debuggers/python/impl.go]: can you expand this comment to the full expression we are looking for?

**Comment by anton-st** [comment]
Fixed

**Pull request merged**